test/widgets/emoji_reaction_test.dart

```diff
@@ -556,6 +556,43 @@ void main() {
         debugNetworkImageHttpClientProvider = null;
       });
 
+      testWidgets('text emoji list items must have correct height', (tester) async {
+      // Regression test for #1587
+      tester.platformDispatcher.textScaleFactorTestValue = 2.0;
+      addTearDown(tester.platformDispatcher.clearTextScaleFactorTestValue);
+
+      final message = eg.streamMessage();
+      // Firstly, set up the initial emoji picker. This leaves it open.
+      await setupEmojiPicker(tester, message: message, narrow: TopicNarrow.ofMessage(message));
+
+      // and Now, change the setting to text-only emoji.
+      await store.handleEvent(UserSettingsUpdateEvent(id: 1,
+        property: UserSettingName.emojiset,
+        value: Emojiset.text));
+
+      // Close the emoji picker. This also closes the underlying action sheet also.
+      Navigator.of(tester.element(find.byType(EmojiPicker))).pop();
+      await tester.pumpAndSettle(); // Let dismiss animations finish.
+
+      // We are now back at the message list. Let's start over to see the setting change.
+      // now we Long-press the message to open the action sheet again.
+      await tester.longPress(find.byType(MessageContent));
+      await tester.pumpAndSettle();
+
+      // now we Tap the button to open the emoji picker, which will now use the new setting.
+      await tester.tap(find.byIcon(ZulipIcons.chevron_right));
+      await tester.pumpAndSettle();
```

Comment:
Is it necessary to close and re-open the emoji picker? What happens if the test doesn't do that?